spec/mongoid/slug_spec.rb

```diff
@@ -471,6 +471,55 @@ module Mongoid
       end
     end
 
+    context 'when block is configured globally' do
+      before do
+        Mongoid::Slug.configure do |c|
+          c.slug = proc{|cur_obj|
```

Comment:
Yes I was trying in this way but I was not getting the block so I. Now I tried it again as you said and realised that its because it was conflicting with the method slug and the slug variable which we use. I need to renamed it. So will use another name and will send a pull request.